pkgs/by-name/uv/uv/package.nix

```diff
@@ -1,19 +1,18 @@
 {
+  stdenv,
   lib,
   cmake,
   fetchFromGitHub,
   installShellFiles,
   pkg-config,
-  python3Packages,
   rustPlatform,
   versionCheckHook,
   nix-update-script,
 }:
 
-python3Packages.buildPythonApplication rec {
+stdenv.mkDerivation rec {
```

Comment:
Why not `rustPlatform.buildRustPackage` then ?

Also, FYI, we did the exact opposite on `ruff` a few days ago: https://github.com/NixOS/nixpkgs/pull/350654.
This is not a problem though.

Reply:
> Why not rustPlatform.buildRustPackage then ?

Because I prefer hooks that are easy to reason about to functional abstractions.

> Also, FYI, we did the exact opposite on ruff a few days ago: https://github.com/NixOS/nixpkgs/pull/350654.

I think that should be undone and aligned with the approach in this PR. It doesn't make sense for packages not attached to a Python package set to have Python libraries.

Reply:
https://github.com/NixOS/nixpkgs/pull/358029

Reply:
I'm not a fan of using a functional package abstractions, especially when the amount of code is very similar to not using one and it obscures what's going on.

It took me a couple of minutes to convert the expression originally. Which hooks needed changing was quick & easy to understand.

Transitioning to `buildRustPackage` I have to understand how it's specific mapping of what's called `cargoDeps` in the previous expression & in the hook is called `cargoLock` with no FOD fetcher applied.
These inconsistencies & papercuts makes me just stick to using plain hooks whenever possible.

I've changed this PR anyway because I don't want it to stall, but I think we should _move away_ from `buildRustPackage` & similar.

Reply:
Ok, this is a fair take. However, this should be discussed in a more general way then. If you feel that using `mkDerivation` for everything (or as most as possible), then you should make a case for it on discourse or in an RFC I guess.
Well, for the packages that you are the maintainer of, I guess that you could do however you want (as long as it satisfies the standard of nixpkgs).
But in general, I think that using the "standards"/"conventions" helps the nixpkgs code-base to remain consistent and not to turn into a zoo of personal preferences.
Also, "you not understanding a hook" is not a reason not to understand it and use it if appropriate.

I might sound harsh and this is not my objective at all. I just wanted to understand better your argument.
It could be completely legit to argue for using `mkDerivation` over `buildRustPackage` in some cases, but here, I was not able to see the benefits.

Best :)